### ISSUE TITLE
Handling Strings without Length

### ISSUE THREAD

**Issue by Fate-JH** [psforever/PSF-LoginServer#67]
Having done some packet diving, I've discovered at least two packets that pass ASCII in a wide character format, but do not have the length prefacing the string directly.  We have no prepared encoding or decoding for this kind of data as they all assume the first byte or pair of bytes are string length, and this means we can not build packet handling logic.  The straightforward aspect of the packet handling also doesn't leave much room for interjection.

For example, a DestroyDisplayMessage packet looks like this:
`81 87`**`41006E00670065006C006C006F00`**`35BC D801 8F 20 19207 0A 0`**`48004D00460049004300`**`B18E D901 00`
There are two character strings here.  For the first string, `87` is the proper length (well, 7 is) so the whole thing can be decoded normally.  For the second string, however, the length is actually encoded in the `0A` in front of the lone `0` as twice the length of the string (5 x 2 = 10, or `0A`; modifying it produces predictable results).  The number format is incorrect, and the buffering `0` just throws the whole thing off.

In addition, passing DestroyDisplayMessage with the correct form of the encoded wide character string to the client - `8548004D00460049004300` - causes the client to display illegible pictograms in place of the string.

Here's an example of a DamageFeedbackMessage:
`7B 76 14`**`48004D00460049004300`**`86 03 90 50 00 00 00`
The wide character string is obvious but, this time, the only suggestions of the size exists as high order bits towards the far end of the data.  Whether or not that's what it is, we can't deal with it.  (A five letter wide character string also has twenty nibbles and `14` is 20 ... I just made that up on the spot.)

**Comment by tfarley** [comment]
The possibility of this certainly exists, and I'm not sure what scodec can do about i.e. purely null-terminated string encoding, but in these cases it's ok. Take a bitstream-oriented approach to break these packets down, where odd strings are _aligned_ wchars (they pad with zero-bits after the string length field to ensure that the actual string contents begin on a byte boundary): http://hastebin.com/azofuyebip.coffee

**Comment by Fate-JH** [comment]
I solved the specific instance of this concern in the contents of the pull request above.  As mentioned, I was talking about a null-terminated string but, as shown, it wasn't one of those.
We don't have need for null-terminated string codec magic yet so this issue can probably be closed.  If I do it, however, I'll wait until after the pull request closes.

**Comment by Fate-JH** [comment]
Found another interesting case - the packet `FriendsResponse`.

Here's an example:
`73 61 8C 38 41 00 6E 00 67 00 65 00 6C 00 6C 00 6F 00 00`
... which turns into:
`73 618C 38 41006E00670065006C006C006F00 00`
... where `618C` was my GUID (confirmed) and the friend is "Angello" as a double-wide 7-character string.  That leaves `38` as the only part of the data portion that can contain the length.
`38` => `0011 1000` => **`00111`**-`000`
It's a __5-bit__ length field with 3-bit padding.

I can show another packet that does the same thing:
`73 61 8C 60 4B 00 75 00 72 00 74 00 48 00 65 00 63 00 74 00 69 00 63 00 2D 00 47 00 00`
`73 618C 60 4B007500720074004800650063007400690063002D004700 00`
String is "KurtHectic-G" and is 12 characters long.
`60` => `0110 0000` => **`01100`**-`000`

I can probably use what I learned from issue #71 and just write new string encoding, by making a custom encoding wrapper around a custom `PacketHelpers.encodedStringSizeWithPad` for a specific `uint(n)` bit-size; but, it's going to be ugly.
